### PR TITLE
Improvement/sentry errors

### LINKED ISSUE #20443: [Task]: Stop reporting non-actionable errors on Sentry
### Is there an existing issue for this?

- [X] I have searched the existing issues

We're exhausting our Sentry quota every month and missing out on the platform-level errors that need attention. Two main contributing factors: 
1. We are tracking non-actionable user-generated errors like [this one](https://appsmith.sentry.io/issues/3829149627/)
2. We are tracking errors on dev environments _(DP, staging, release, etc...)_ that are not so useful

### PR DESCRIPTION
**Pull request by Druthi**
## Description
Stop reporting non-actionable errors on Sentry

Fixes #20443 

## Type of change
- Chore (housekeeping or task changes that don't impact user perception)

## How Has This Been Tested?
- Manual

### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

**Comment by Druthi** [comment]
@Rishabh-Rathod It's a bit tough to reproduce some of the errors reported on the card. So I put checks to ensure there is no errors. We want to ensure user errors are also not picked up. So I tried to reproduce them but it seems we are already not picking up these errors - 

api response error
query response error
Data source connection
Lint

Only time we were passing the errors to sentry was for the Cyclic dependency error which I removed.

File: app/client/src/workers/Evaluation/validations.ts [hunk 1]
```diff
@@ -322,7 +322,12 @@ export const validate = (
     propertyPath,
   );
 
-  return _result;
+  return (
+    _result || {
+      isValid: true,
+      parsed: value,
+    }
+  );
```
Comment (Rishabh-Rathod) [inline app/client/src/workers/Evaluation/validations.ts:330]: There are few cases when `validate` receive an invalid `config.type`.  

Does the below code make sense to handle those cases too?  

```
const validationFn = VALIDATORS[config.type];
const validResponse = {
      isValid: true,
      parsed: value,
    }
if(!validationFn) return validResponse;
return validationFn() || validResponse;
 ```

File: app/client/src/sagas/PostEvaluationSagas.ts [hunk 2]
```diff
@@ -225,27 +225,15 @@ export function* evalErrorHandler(
...
           });
           AppsmithConsole.error({
             text: `${error.message} Node was: ${node}`,
           });
-          // Send the generic error message to sentry for better grouping
-          Sentry.captureException(new Error(error.message), {
-            tags: {
-              node,
-              entityType,
-            },
```
Comment (Rishabh-Rathod) [inline app/client/src/sagas/PostEvaluationSagas.ts:241]: `entityType` and `node` have been helpful in debugging. Any reason why we are removing them?

Comment (Druthi) [reply app/client/src/sagas/PostEvaluationSagas.ts:241]: I was wondering the same. Let me clarify with Bharath and get back

**Pull request closed without merging**